parser-typechecker/src/Unison/Codebase/SqliteCodebase/Operations.hs

```diff
@@ -599,13 +600,50 @@ namesAtPath path = do
         Nothing -> error $ "Expected name to be in namespace" <> show (n, reverse reversedPathSegments)
         Just stripped -> (Name.makeRelative stripped, ref)
 
+-- | Update the root namespace names index which is used by the share server for serving api
+-- requests.
+updateNameLookupIndex ::
+  (C.Reference.Reference -> Sqlite.Transaction CT.ConstructorType) ->
+  Path ->
+  -- | "from" branch, if 'Nothing' use the empty branch
+  Maybe BranchHash ->
+  -- | "to" branch, if 'Nothing' use the root.
+  Maybe BranchHash ->
```

Comment:
Minor suggestion: make the caller provide the 'to' Branch hash, just so this function has fewer things to do.